### PR TITLE
Adapt Gammapy to regions v0.5

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This pull request adapts Gammapy to rely on `region=v0.5`. The most significant change is a serialisation API on `Regions` (see https://astropy-regions.readthedocs.io/en/stable/changelog.html#new-features). I adapted Gammapy to rely on this and decided to remove our custom `make_region` methods. The pattern is now:
```python3
region = Regions.parse("icrc;circle(0, 0, 1)", format="ds9")
```
This is a bit more verbose than the `make_region` methods, but much more flexible. It allows to also create multiple regions and not a single one and support multiple formats. In general it is always preferable to rely on the external API if possible. I also adapted `RegionGeom.create()` so one can now pass a region string containing multiple regions.

### CODE REVIEW

**Review by registerrier — COMMENTED** [review]
Thanks @adonath . This looks good.

Should we fully remove `make_region` ? I clearly see that it does less that the new `Regions.Parse`. Yet the latter is a feature we have advertised because it does not require knowledge of the regions package. 
How likely is the regions interface to change in the future? Keeping the `maker_region` would allow at least to have a stable way to create regions during the v1.0 lifetime.

**Comment by adonath** [comment]
@registerrier I'm in favour of not supporting a new API for region creation. `regions` is a required dependency to Gammapy and users should at least know the basic of it. However we should make sure, that everywhere in Gammapy where we take a `region` argument we go through `RegionGeom.create().region` after, to always allow user to pass as string to a method.

File: gammapy/datasets/tests/test_spectrum.py [hunk 1]
```diff
@@ -515,6 +511,8 @@ def test_to_from_ogip_files(self, tmp_path):
         assert_allclose(self.edisp.edisp_map.data, newdataset.edisp.edisp_map.data)
         assert_time_allclose(newdataset.gti.time_start, dataset.gti.time_start)
 
+        #print(self.off_counts.geom.region)
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/datasets/tests/test_spectrum.py]: Done

File: gammapy/maps/region.py [hunk 2]
```diff
@@ -493,24 +492,21 @@ def _make_bands_cols(self):
...
-        """Create region.
+        """Create region geometry.
 
         Parameters
         ----------
         region : str or `~regions.SkyRegion`
-            Region
-        axes : list of `MapAxis`
-            Non spatial axes.
+            Region definition
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/maps/region.py:500]: Let's resolve it in a uniform way, together with a recommendation in the developer docs.

**Comment by adonath** [comment]
@registerrier and @bkhelifi I changed my mind here: keeping `make_region` now, means we keep it for v1.0. It does not make any sense to show users `make_region` in one place and `Regions.parse()` in another place. Yes, old users have to adapt, but new users (which we'll get many hopefully with v1.0...) will learn a single nice and uniform API. I agree something like:

```python3
region = Regions.parse("icrc;circle(0, 0, 1)", format="ds9")[0]
```
Is a bit less nice, but if users don't understand the meaning of a simple indexing `[0]`, then they should not use the 2nd level API at all but rely on the high level interface instead. In the mask tutorial one can even ignore the `[0]`, because all functions in Gammapy take a list fo regions as well.

The `make_region()` API does not provide any additional value. The argument that if the API in `regions` changes ist not really valid, because it would mean to wrap all the functionality in `regions` in our own API, which is clearly not an option. Previously `regions` didn't have a nice API to create regions from DS9 string, now there is a better and more flexibel one and we can just rely on it.

The API that we fully control is the `RegionGeom` instead and this will remain. However, as I mentioned in  https://github.com/gammapy/gammapy/pull/3453#issuecomment-893559126, we should make sure that everywhere were we expect a `region` argument, we call `RegionGeom.create().region` after to allow for uniform and convenient ds9 string input.

**Comment by adonath** [comment]
@bkhelifi and @registerrier Just to extend on it: if we allow the string input everywhere users will never need the `make_region` anyway, because the region is created to be passed to any Gammapy method. This is exactly in line with [PIG 10](https://docs.gammapy.org/0.18.2/development/pigs/pig-010.html#region-arguments), where there is even the statement:
> We suggest to start by implementing these two helper functions in gammapy/utils/regions.py and to use it internally, but not expose it as part of the public API via the Gammapy docs. 

So `make_region` was never meant to part of the public API...

**Comment by adonath** [comment]
I adapted the `mask_maps.ipynb` tutorial to directly use strings if a single region is passed. This corresponds to the intended use outlined in PIG-10.

**Pull request merged**